src/components/UI/Spinner/Spinner.jsx

```diff
@@ -2,7 +2,7 @@ import { useState, useEffect } from "react";
 import { PulseLoader } from "react-spinners";
 import "./Spinner.css";
 
-const Spinner = (props) => {
+const Spinner = (Component) => {
```

Comment:
Checked.
Here is the root cause: since you are using your HOC as a function that gets your component as argument and returns extended version of it, React is claiming your HOC as a function, not a functional component to begin with. this is why you are having such kind of issue. 
The solution is pretty simple - you need to return functional component as a result of working of your function by just using extra return and calling your HOC with lowercase letter:

```
import { useState, useEffect } from "react";
import { PulseLoader } from "react-spinners";
import "./Spinner.css";

const spinner = (Component) => props => {
  const [showSpinner, setShowSpinner] = useState(true);

  useEffect(() => {
    setTimeout(() => {
      setShowSpinner(false);
    }, 2000);
  }, []);

  return showSpinner ? (
    <div className="spinner-card">
        <PulseLoader color="rgb(34, 4, 116)" size={20} margin={7}></PulseLoader>
    </div>
  ) : (
    <Component {...props}/>
  );
};

export default spinner;
```
Please check, this should work.